Installation issue: xorg-server

### Steps to reproduce the issue

```console
$ spack spec -I xorg-server
Input spec
--------------------------------
 -   xorg-server

Concretized
--------------------------------
 -   xorg-server@1.18.99.901%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[+]      ^bigreqsproto@1.1.2%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[^]      ^bison@3.8.2%gcc@8.5.0 arch=linux-rocky8-zen
[e]          ^diffutils@3.6%gcc@8.5.0 arch=linux-rocky8-zen
[e]          ^m4@1.4.18%gcc@8.5.0+sigsegv patches=3877ab5,fc9b616 arch=linux-rocky8-zen
[^]          ^perl@5.34.1%gcc@8.5.0+cpanm+shared+threads arch=linux-rocky8-zen
[^]              ^berkeley-db@18.1.40%gcc@8.5.0+cxx~docs+stl patches=b231fcc arch=linux-rocky8-zen
[^]              ^bzip2@1.0.8%gcc@8.5.0~debug~pic+shared arch=linux-rocky8-zen
[^]              ^gdbm@1.19%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^readline@8.1%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^zlib@1.2.12%gcc@8.5.0+optimize+pic+shared patches=0d38234 arch=linux-rocky8-zen
[+]      ^compositeproto@0.4.2%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[+]      ^damageproto@1.2.1%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[+]      ^dri2proto@2.8%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[+]      ^dri3proto@1.0%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[^]      ^fixesproto@5.0%gcc@8.5.0 arch=linux-rocky8-zen
[e]      ^flex@2.6.1%gcc@8.5.0+lex~nls arch=linux-rocky8-zen
[^]      ^font-util@1.3.2%gcc@8.5.0 fonts=encodings,font-adobe-100dpi,font-adobe-75dpi,font-adobe-utopia-100dpi,font-adobe-utopia-75dpi,font-adobe-utopia-type1,font-alias,font-arabic-misc,font-bh-100dpi,font-bh-75dpi,font-bh-lucidatypewriter-100dpi,font-bh-lucidatypewriter-75dpi,font-bh-type1,font-bitstream-100dpi,font-bitstream-75dpi,font-bitstream-speedo,font-bitstream-type1,font-cronyx-cyrillic,font-cursor-misc,font-daewoo-misc,font-dec-misc,font-ibm-type1,font-isas-misc,font-jis-misc,font-micro-misc,font-misc-cyrillic,font-misc-ethiopic,font-misc-meltho,font-misc-misc,font-mutt-misc,font-schumacher-misc,font-screen-cyrillic,font-sun-misc,font-winitzki-cyrillic,font-xfree86-type1 arch=linux-rocky8-zen
[e]          ^autoconf@2.69%gcc@8.5.0 patches=7793209 arch=linux-rocky8-zen
[e]          ^automake@1.16.1%gcc@8.5.0 arch=linux-rocky8-zen
[^]          ^bdftopcf@1.0.5%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^libxfont@1.5.2%gcc@8.5.0 arch=linux-rocky8-zen
[^]          ^mkfontdir@1.0.7%gcc@8.5.0 arch=linux-rocky8-zen
[^]          ^mkfontscale@1.1.2%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^fontsproto@2.1.3%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^glproto@1.4.17%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^inputproto@2.3.2%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^kbproto@1.0.7%gcc@8.5.0 arch=linux-rocky8-zen
[+]      ^libdrm@2.4.115%gcc@8.5.0~docs build_system=generic arch=linux-rocky8-zen
[+]          ^libpciaccess@0.17%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[e]              ^libtool@2.4.6%gcc@8.5.0 arch=linux-rocky8-zen
[^]          ^libpthread-stubs@0.4%gcc@8.5.0 arch=linux-rocky8-zen
[^]          ^meson@0.62.1%gcc@8.5.0 patches=aa6c50d arch=linux-rocky8-zen
[^]              ^ninja@1.10.2%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^py-pip@21.3.1%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^py-setuptools@59.4.0%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^py-wheel@0.37.0%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^python@3.7.13%gcc@8.5.0+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,f2fd060 arch=linux-rocky8-zen
[^]                  ^libffi@3.4.2%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^ncurses@6.2%gcc@8.5.0~symlinks+termlib abi=none arch=linux-rocky8-zen
[e]                  ^openssl@1.1.1k%gcc@8.5.0~docs~shared certs=system arch=linux-rocky8-zen
[^]                  ^sqlite@3.38.5%gcc@8.5.0+column_metadata+dynamic_extensions+fts~functions+rtree arch=linux-rocky8-zen
[^]                  ^util-linux-uuid@2.37.4%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^xz@5.2.5%gcc@8.5.0~pic libs=shared,static arch=linux-rocky8-zen
[^]      ^libepoxy@1.4.3%gcc@8.5.0+glx arch=linux-rocky8-zen
[^]          ^mesa@22.0.2%gcc@8.5.0+glx+llvm+opengl~opengles+osmesa~strip buildtype=release default_library=shared patches=ada85be arch=linux-rocky8-zen
[e]              ^cmake@3.20.2%gcc@8.5.0~doc+ncurses+ownlibs~qt build_type=Release arch=linux-rocky8-zen
[^]              ^expat@2.4.8%gcc@8.5.0+libbsd arch=linux-rocky8-zen
[^]                  ^libbsd@0.11.5%gcc@8.5.0 arch=linux-rocky8-zen
[^]                      ^libmd@1.0.4%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^gettext@0.21%gcc@8.5.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-rocky8-zen
[^]                  ^libiconv@1.16%gcc@8.5.0 libs=shared,static arch=linux-rocky8-zen
[e]                  ^libxml2@2.9.7%gcc@8.5.0~python arch=linux-rocky8-zen
[e]                  ^tar@1.30%gcc@8.5.0 zip=pigz arch=linux-rocky8-zen
[^]              ^libunwind@1.6.2%gcc@8.5.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib components=none libs=shared,static arch=linux-rocky8-zen
[^]              ^libxt@1.1.5%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^libice@1.0.9%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^libsm@1.2.3%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^llvm@12.0.1%gcc@8.5.0+clang~code_signing+compiler-rt~cuda~flang+gold+internal_unwind~ipo+libcxx~link_llvm_dylib+lld+lldb~llvm_dylib~mlir+omp_as_runtime~omp_debug~omp_tsan+polly~python~shared_libs~split_dwarf~z3 build_type=Release patches=f920173 targets=none version_suffix=none arch=linux-rocky8-zen
[e]                  ^binutils@2.30.113%gcc@8.5.0~gas+gold~headers~interwork+ld~libiberty~lto+nls+plugins libs=shared,static arch=linux-rocky8-zen
[e]                  ^hwloc@2.2.0%gcc@8.5.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~opencl+pci~rocm+shared arch=linux-rocky8-zen
[^]                  ^libedit@3.1-20210216%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^perl-data-dumper@2.173%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^swig@4.0.2%gcc@8.5.0 arch=linux-rocky8-zen
[^]                      ^pcre@8.45%gcc@8.5.0~jit+multibyte+utf arch=linux-rocky8-zen
[^]              ^meson@0.62.1%gcc@8.5.0 patches=aa6c50d arch=linux-rocky8-zen
[^]                  ^py-pip@21.3.1%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^py-setuptools@59.4.0%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^py-wheel@0.37.0%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^py-mako@1.1.6%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^py-markupsafe@2.0.1%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^python@3.9.12%gcc@8.5.0+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,f2fd060 arch=linux-rocky8-zen
[^]              ^xrandr@1.5.0%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^libxrandr@1.5.0%gcc@8.5.0 arch=linux-rocky8-zen
[^]                  ^libxrender@0.9.10%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^libx11@1.7.0%gcc@8.5.0 arch=linux-rocky8-zen
[^]          ^libxcb@1.14%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^libxau@1.0.8%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^libxdmcp@1.1.2%gcc@8.5.0 arch=linux-rocky8-zen
[^]              ^xcb-proto@1.14.1%gcc@8.5.0 arch=linux-rocky8-zen
[+]      ^libxdamage@1.1.4%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[^]      ^libxext@1.3.3%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^libxfixes@5.0.2%gcc@8.5.0 arch=linux-rocky8-zen
[+]      ^libxfont2@2.0.6%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[^]          ^freetype@2.11.1%gcc@8.5.0 arch=linux-rocky8-zen
[^]          ^libfontenc@1.1.3%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^libxkbfile@1.0.9%gcc@8.5.0 arch=linux-rocky8-zen
[+]      ^libxshmfence@1.3%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[^]      ^pixman@0.40.0%gcc@8.5.0 arch=linux-rocky8-zen
[e]          ^libpng@1.6.34%gcc@8.5.0 arch=linux-rocky8-zen
[e]      ^pkgconf@1.4.2%gcc@8.5.0 arch=linux-rocky8-zen
[+]      ^presentproto@1.1%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[^]      ^randrproto@1.5.0%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^recordproto@1.14.2%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^renderproto@0.11.1%gcc@8.5.0 arch=linux-rocky8-zen
[+]      ^resourceproto@1.2.0%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[^]      ^scrnsaverproto@1.2.2%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^util-macros@1.19.3%gcc@8.5.0 arch=linux-rocky8-zen
[+]      ^videoproto@2.3.3%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[+]      ^xcmiscproto@1.2.2%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[^]      ^xextproto@7.3.0%gcc@8.5.0 arch=linux-rocky8-zen
[+]      ^xf86driproto@2.1.1%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[+]      ^xineramaproto@1.2.1%gcc@8.5.0 build_system=autotools arch=linux-rocky8-zen
[^]      ^xproto@7.0.31%gcc@8.5.0 arch=linux-rocky8-zen
[^]      ^xtrans@1.3.5%gcc@8.5.0 arch=linux-rocky8-zen
```


### Error message

There seems to be a problem correctly installing the dependencies. The package 'dri' cannot be found.

<details><summary>Error message</summary>
The error is: Package 'dri', required by 'virtual:world', not found
<pre>
[+] /albedo/home/pgierz/.spack/sw/bigreqsproto/1.1.2-fkgg7ck
[+] /usr (external m4-1.4.18-6g6dyn6gyt4jd2aldglok4lo5ypa7tsi)
[+] /albedo/home/pgierz/.spack/sw/compositeproto/0.4.2-rt73zml
[+] /albedo/home/pgierz/.spack/sw/damageproto/1.2.1-jbotgjm
[+] /albedo/home/pgierz/.spack/sw/dri2proto/2.8-u6dtggl
[+] /albedo/home/pgierz/.spack/sw/dri3proto/1.0-xkkbgvt
[+] /albedo/soft/sw/spack-sw/fixesproto/5.0-d47an4b
[+] /usr (external flex-2.6.1-6tjs7yimxl6lcplgllifvh57azfpfune)
[+] /albedo/soft/sw/spack-sw/font-util/1.3.2-p6dsv6j
[+] /albedo/soft/sw/spack-sw/fontsproto/2.1.3-yaimsvs
[+] /albedo/soft/sw/spack-sw/glproto/1.4.17-k2mwuv6
[+] /albedo/soft/sw/spack-sw/inputproto/2.3.2-n25zulq
[+] /albedo/soft/sw/spack-sw/kbproto/1.0.7-ecphj2c
[+] /albedo/home/pgierz/.spack/sw/libpciaccess/0.17-5yohxaq
[+] /albedo/soft/sw/spack-sw/libpthread-stubs/0.4-gnmwtes
[+] /albedo/soft/sw/spack-sw/xproto/7.0.31-ssfmh2d
[+] /albedo/soft/sw/spack-sw/libmd/1.0.4-32hgrgf
[+] /albedo/soft/sw/spack-sw/xcb-proto/1.14.1-2pa37of
[+] /albedo/soft/sw/spack-sw/xextproto/7.3.0-nwec36b
[+] /albedo/soft/sw/spack-sw/xtrans/1.3.5-7emcyln
[+] /albedo/soft/sw/spack-sw/libunwind/1.6.2-p6n4rb7
[+] /albedo/soft/sw/spack-sw/util-linux-uuid/2.37.4-lfk5gdi
[+] /usr (external binutils-2.30.113-wtvavja4hyq2j5fq2a5og2w4p2smznbq)
[+] /usr (external hwloc-2.2.0-smlq4qrbxkhitjejdy4mo6bocozrc4xq)
[+] /albedo/soft/sw/spack-sw/ncurses/6.2-2phlyvl
[+] /usr (external libxml2-2.9.7-yepp3ed7c4q6epmsmssz7cqrlcsf7n6h)
[+] /albedo/soft/sw/spack-sw/pcre/8.45-36x3ei2
[+] /albedo/soft/sw/spack-sw/zlib/1.2.12-txishb5
[+] /albedo/soft/sw/spack-sw/renderproto/0.11.1-rwfv3d2
[+] /albedo/soft/sw/spack-sw/randrproto/1.5.0-hyh245m
[+] /albedo/soft/sw/spack-sw/bzip2/1.0.8-tc3c42c
[+] /usr (external libpng-1.6.34-znvmzzndcselwa34ffrrayk7y6feiwhj)
[+] /usr (external pkgconf-1.4.2-t7zbaxvugfxkrwcr5hbizm2kuqjyue2n)
[+] /albedo/soft/sw/spack-sw/util-macros/1.19.3-3zhsk4y
[+] /albedo/soft/sw/spack-sw/bison/3.8.2-5a463xm
[+] /albedo/soft/sw/spack-sw/libbsd/0.11.5-5spk7z7
[+] /albedo/soft/sw/spack-sw/swig/4.0.2-3g4yaqq
[+] /albedo/soft/sw/spack-sw/libedit/3.1-20210216-ntyqrny
[+] /albedo/soft/sw/spack-sw/freetype/2.11.1-l7umill
[+] /albedo/soft/sw/spack-sw/pixman/0.40.0-ujllftx
[+] /albedo/home/pgierz/.spack/sw/libdrm/2.4.115-c3oq3w3
[+] /albedo/home/pgierz/.spack/sw/xcmiscproto/1.2.2-bajrw5e
[+] /albedo/home/pgierz/.spack/sw/libxshmfence/1.3-m6dmsnp
[+] /albedo/soft/sw/spack-sw/scrnsaverproto/1.2.2-j55vbsw
[+] /albedo/home/pgierz/.spack/sw/resourceproto/1.2.0-lhzvlgr
[+] /albedo/soft/sw/spack-sw/libfontenc/1.1.3-e23eesq
[+] /albedo/home/pgierz/.spack/sw/xineramaproto/1.2.1-qqki2u6
[+] /albedo/soft/sw/spack-sw/libice/1.0.9-dsnunrv
[+] /albedo/home/pgierz/.spack/sw/presentproto/1.1-qsxslks
[+] /albedo/home/pgierz/.spack/sw/videoproto/2.3.3-rsdpzjb
[+] /albedo/soft/sw/spack-sw/libxau/1.0.8-kg6jqeh
[+] /albedo/home/pgierz/.spack/sw/xf86driproto/2.1.1-o3y7i2u
[+] /albedo/soft/sw/spack-sw/recordproto/1.14.2-ogifi6i
[+] /albedo/soft/sw/spack-sw/expat/2.4.8-km7l2qr
[+] /albedo/soft/sw/spack-sw/libxdmcp/1.1.2-pepegea
[+] /albedo/soft/sw/spack-sw/llvm/12.0.1-ytq7ixt
[+] /albedo/home/pgierz/.spack/sw/libxfont2/2.0.6-huepnb6
[+] /albedo/soft/sw/spack-sw/libsm/1.2.3-ndoc74u
[+] /albedo/soft/sw/spack-sw/libxcb/1.14-mkgui3l
[+] /albedo/soft/sw/spack-sw/libx11/1.7.0-6j3srlt
[+] /albedo/soft/sw/spack-sw/libxrender/0.9.10-jrk27wx
[+] /albedo/soft/sw/spack-sw/libxt/1.1.5-23xlbjc
[+] /albedo/soft/sw/spack-sw/libxfixes/5.0.2-fk4njnn
[+] /albedo/soft/sw/spack-sw/libxkbfile/1.0.9-vhy6dvt
[+] /albedo/soft/sw/spack-sw/libxext/1.3.3-rejaqxr
[+] /albedo/home/pgierz/.spack/sw/libxdamage/1.1.4-qssfkyl
[+] /albedo/soft/sw/spack-sw/libxrandr/1.5.0-xmknoaw
[+] /albedo/soft/sw/spack-sw/xrandr/1.5.0-fodtvmx
[+] /albedo/soft/sw/spack-sw/mesa/22.0.2-mmlm5co
[+] /albedo/soft/sw/spack-sw/libepoxy/1.4.3-jcjukhk
==> Installing xorg-server-1.18.99.901-4k26jmwnvjxgp6z2fle623kwq35kf4pl
==> No binary for xorg-server-1.18.99.901-4k26jmwnvjxgp6z2fle623kwq35kf4pl found: installing from source
==> Using cached archive: /albedo/home/pgierz/.spack/cache/source/_source-cache/archive/c8/c8425163b588de2ee7e5c8e65b0749f2710f55a7e02a8d1dc83b3630868ceb21.tar.gz
==> No patches needed for xorg-server
==> xorg-server: Executing phase: 'autoreconf'
==> xorg-server: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/tmp/pgierz/spack-stage/spack-stage-xorg-server-1.18.99.901-4k26jmwnvjxgp6z2fle623kwq35kf4pl/spack-src/configure' '--prefix=/albedo/home/pgierz/.spack/sw/xorg-server/1.18.99.901-4k26jmw'

1 error found in build log:
     321    checking for nettle_sha1_init in -lnettle... no
     322    checking for gcry_md_open in -lgcrypt... no
     323    checking for SHA1_Init in -lcrypto... yes
     324    checking for OPENSSL... yes
     325    checking for SHA1 implementation... libmd
     326    checking for XSERVERCFLAGS... no
  >> 327    configure: error: Package requirements (fixesproto >= 5.0 damageproto >= 1.1 xcmiscproto >= 1.2.0 xtrans >= 1.3.5 bigreqsproto >= 1.1.0 xproto >=
             7.0.28 randrproto >= 1.5.0 renderproto >= 0.11 xextproto >= 7.2.99.901 inputproto >= 2.3 kbproto >= 1.0.3 fontsproto >= 2.1.3 pixman-1 >= 0.27.2
             videoproto compositeproto >= 0.4 recordproto >= 1.13.99.1 scrnsaverproto >= 1.1 resourceproto >= 1.2.0 xf86driproto >= 2.1.0 glproto >= 1.4.17 d
            ri >= 7.8.0 presentproto >= 1.0 xineramaproto xkbfile  pixman-1 >= 0.27.2 xfont2 >= 2.0.0 xau xshmfence >= 1.1 xdmcp) were not met:
     328
     329    Package 'dri', required by 'virtual:world', not found
     330
     331    Consider adjusting the PKG_CONFIG_PATH environment variable if you
     332    installed software in a non-standard prefix.
     333
</pre></details>


### Information on your system

spack debug report
* **Spack:** 0.20.1 (e8658d6493887ef702dd38f0e9ee5870a1651c1e)
* **Python:** 3.10.4
* **Platform:** linux-rocky8-zen2
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/user-attachments/files/16613992/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/16613993/spack-build-out.txt)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/xorg-server/package.py
+++ b/var/spack/repos/builtin/packages/xorg-server/package.py
@@ -19,6 +19,9 @@ class XorgServer(AutotoolsPackage, XorgPackage):
         "1.18.99.901", sha256="c8425163b588de2ee7e5c8e65b0749f2710f55a7e02a8d1dc83b3630868ceb21"
     )
 
+    variant("glx", default=True, description="Build GLX extension")
+    variant("dri", default=True, description="Build DRI, DRI2, DRI3 extensions")
+
     # glibc stopped declaring major()/minor() macros in <sys/types.h>
     # https://gitlab.freedesktop.org/xorg/xserver/-/commit/d732c36597fab2e9bc4f2aa72cf1110997697557
     patch("sysmacros.patch", when="@:1.18 ^glibc@2.25:")
@@ -38,11 +41,14 @@ class XorgServer(AutotoolsPackage, XorgPackage):
     depends_on("libdrm@2.3.0:")
     depends_on("libx11")
 
-    depends_on("gl")
+    depends_on("gl", when="+dri")
+    depends_on("gl", when="+glx")
 
-    depends_on("dri2proto@2.8:", type="build")
-    depends_on("dri3proto@1.0:", type="build")
-    depends_on("glproto@1.4.17:", type="build")
+    depends_on("xf86driproto@2.1.0:", type="build", when="+dri")
+    depends_on("dri2proto@2.8:", type="build", when="+dri")
+    depends_on("dri3proto@1.0:", type="build", when="+dri")
+    depends_on("glproto@1.4.17:", type="build", when="+dri")
+    depends_on("glproto@1.4.17:", type="build", when="+glx")
 
     depends_on("flex", type="build")
     depends_on("bison", type="build")
@@ -66,7 +72,6 @@ class XorgServer(AutotoolsPackage, XorgPackage):
     depends_on("recordproto@1.13.99.1:", type="build")
     depends_on("scrnsaverproto@1.1:", type="build")
     depends_on("resourceproto@1.2.0:", type="build")
-    depends_on("xf86driproto@2.1.0:", type="build")
     depends_on("glproto@1.4.17:", type="build")
     depends_on("presentproto@1.0:", type="build")
     depends_on("xineramaproto", type="build")
@@ -88,6 +93,20 @@ class XorgServer(AutotoolsPackage, XorgPackage):
 
     def configure_args(self):
         args = []
+
+        if self.spec.satisfies("+glx ^[virtuals=gl] osmesa"):
+            args.append("--enable-glx")
+        else:
+            args.append("--disable-glx")
+            
+        if self.spec.satisfies("+dri"):
+            args.append("--enable-dri")
+            args.append("--enable-dri2")
+            args.append("--enable-dri3")
+        else:
+            args.append("--disable-dri")
+            args.append("--disable-dri2")
+            args.append("--disable-dri3")
 
         if self.spec.satisfies("^[virtuals=gl] osmesa"):
             args.append("--enable-glx")

--- a/var/spack/repos/builtin/packages/xorg-server/package.py
+++ b/var/spack/repos/builtin/packages/xorg-server/package.py
@@ -16,6 +16,9 @@ class XorgServer(AutotoolsPackage, XorgPackage):
     license("MIT")
 
     version(
+        "1.18.99.902", sha256="fe5a312f7bdc6762c97f01b7a1d3c7a8691997255be6fbf7598c180abf384ea3"
+    )
+    version(
         "1.18.99.901", sha256="c8425163b588de2ee7e5c8e65b0749f2710f55a7e02a8d1dc83b3630868ceb21"
     )
 

--- a/var/spack/repos/builtin/packages/xorg-server/package.py
+++ b/var/spack/repos/builtin/packages/xorg-server/package.py
@@ -41,7 +41,7 @@ class XorgServer(AutotoolsPackage, XorgPackage):
     depends_on("pixman@0.27.2:")
     depends_on("font-util")
     depends_on("libxshmfence@1.1:")
-    depends_on("libdrm@2.3.0:")
+    depends_on("libdrm@2.3.0:", when="+dri")
     depends_on("libx11")
 
     depends_on("gl", when="+dri")

--- a/var/spack/repos/builtin/packages/xorg-server/package.py
+++ b/var/spack/repos/builtin/packages/xorg-server/package.py
@@ -103,7 +103,7 @@ class XorgServer(AutotoolsPackage, XorgPackage):
             args.append("--disable-glx")
 
         if self.spec.satisfies("+dri"):
-            args.append("--enable-dri")
+            args.append("--disable-dri")  # dri requires libdri, not libdrm
             args.append("--enable-dri2")
             args.append("--enable-dri3")
             args.append("--enable-drm")

--- a/var/spack/repos/builtin/packages/xorg-server/package.py
+++ b/var/spack/repos/builtin/packages/xorg-server/package.py
@@ -75,7 +75,6 @@ class XorgServer(AutotoolsPackage, XorgPackage):
     depends_on("recordproto@1.13.99.1:", type="build")
     depends_on("scrnsaverproto@1.1:", type="build")
     depends_on("resourceproto@1.2.0:", type="build")
-    depends_on("glproto@1.4.17:", type="build")
     depends_on("presentproto@1.0:", type="build")
     depends_on("xineramaproto", type="build")
     depends_on("libxkbfile")
@@ -84,6 +83,7 @@ class XorgServer(AutotoolsPackage, XorgPackage):
     depends_on("libxdamage")
     depends_on("libxfixes")
     depends_on("libepoxy")
+    depends_on("libpciaccess")
 
     @when("@:1.19")
     def setup_build_environment(self, env):
@@ -106,10 +106,12 @@ class XorgServer(AutotoolsPackage, XorgPackage):
             args.append("--enable-dri")
             args.append("--enable-dri2")
             args.append("--enable-dri3")
+            args.append("--enable-drm")
         else:
             args.append("--disable-dri")
             args.append("--disable-dri2")
             args.append("--disable-dri3")
+            args.append("--disable-drm")
 
         if self.spec.satisfies("^[virtuals=gl] osmesa"):
             args.append("--enable-glx")
@@ -118,7 +120,6 @@ class XorgServer(AutotoolsPackage, XorgPackage):
 
         args.extend(
             [
-                "--disable-dri",  # dri >= 7.8.0
                 "--disable-glamor",  # Glamor for Xorg requires gbm >= 10.2.0
             ]
         )

--- a/var/spack/repos/builtin/packages/xorg-server/package.py
+++ b/var/spack/repos/builtin/packages/xorg-server/package.py
@@ -101,7 +101,7 @@ class XorgServer(AutotoolsPackage, XorgPackage):
             args.append("--enable-glx")
         else:
             args.append("--disable-glx")
-            
+
         if self.spec.satisfies("+dri"):
             args.append("--enable-dri")
             args.append("--enable-dri2")
@@ -118,10 +118,6 @@ class XorgServer(AutotoolsPackage, XorgPackage):
         else:
             args.append("--disable-glx")
 
-        args.extend(
-            [
-                "--disable-glamor",  # Glamor for Xorg requires gbm >= 10.2.0
-            ]
-        )
+        args.extend(["--disable-glamor"])  # Glamor for Xorg requires gbm >= 10.2.0
 
         return args

